tests/integration/src/dfx_orbit/assets.rs

```diff
@@ -0,0 +1,155 @@
+            CallExternalCanisterOperationInput {
+                validation_method: None,
+                execution_method: CanisterMethodDTO {
+                    canister_id: asset_canister,
+                    method_name: String::from("grant_permission"),
+                },
+                arg: Some(StationAgent::request_prepare_permission_payload(dfx_principal).unwrap()),
+                execution_method_cycles: None,
+            },
+        ),
+    )
+    .unwrap();
+
+    // As admin: Grant the user the call permission, set auto-approval for external calls
+    permit_call_operation(&env, &canister_ids);
+    set_auto_approve(&env, &canister_ids);
+
+    // Setup a tmpdir, and store two assets in it
+    // We generate the assets dyniamically, since we want to make sure we are not
+    // fetching old assets
+    let asset_dir = tempdir().unwrap();
+    let asset_a = format!(
+        "This is the current time: {}",
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_nanos()
+    );
+    let asset_b = format!("This is a random number: {}", thread_rng().gen::<u64>());
+
+    std::fs::create_dir_all(asset_dir.path().join("subdir")).unwrap();
+    std::fs::write(asset_dir.path().join("system_time"), &asset_a).unwrap();
+    std::fs::write(
+        asset_dir.path().join("subdir").join("random_number"),
+        &asset_b,
+    )
+    .unwrap();
+
+    dfx_orbit_test(&mut env, async {
+        // Setup the station agent
+        let mut dfx_orbit = setup_dfx_orbit(canister_ids.station).await;
+
+        // As dfx user: Request to upload new files to the asset canister
+        let upload_request = dfx_orbit
+            .upload_assets(
+                asset_canister.to_string(),
+                vec![asset_dir.path().to_str().unwrap().to_string()],
+            )
+            .await
+            .unwrap();
+
+        //  As dfx user: Request commitment of the batch
+        let _result = dfx_orbit
+            .station
+            .request(CreateRequestInput {
+                operation: station_api::RequestOperationInput::CallExternalCanister(
+                    CallExternalCanisterOperationInput {
+                        validation_method: None,
+                        execution_method: CanisterMethodDTO {
+                            canister_id: asset_canister,
+                            method_name: String::from("commit_proposed_batch"),
+                        },
+                        arg: Some(
+                            StationAgent::commit_proposed_batch_payload(upload_request).unwrap(),
+                        ),
+                        execution_method_cycles: None,
+                    },
+                ),
+                title: None,
+                summary: None,
+                execution_plan: None,
+            })
+            .await
+            .unwrap();
+        tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
any reason to sleep instead of doing a given number of ticks?

Reply:
I'll add a comment.

Here, the PocketIC is in "live" mode, so it uses realtime to advance the ticks.
If we leave life mode, I think we can no longer access the http enpoint to make the request.

I can add a comment. Also I wanted to figure out a way to switch back and forth between "live" mode and not live mode, but at the moment, we can only go live once.